src/Servers/Connections.Abstractions/src/NamedPipeEndPoint.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+
+namespace Microsoft.AspNetCore.Connections;
+
+/// <summary>
+/// Represents a Named Pipe endpoint.
+/// </summary>
+public sealed class NamedPipeEndPoint : EndPoint
+{
+    internal const string LocalComputerServerName = ".";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NamedPipeEndPoint"/> class.
+    /// </summary>
+    /// <param name="pipeName">The name of the pipe.</param>
+    public NamedPipeEndPoint(string pipeName) : this(pipeName, LocalComputerServerName)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NamedPipeEndPoint"/> class.
+    /// </summary>
+    /// <param name="pipeName">The name of the pipe.</param>
+    /// <param name="serverName">The name of the remote computer to connect to, or "." to specify the local computer.</param>
+    public NamedPipeEndPoint(string pipeName, string serverName)
+    {
+        ServerName = serverName;
+        PipeName = pipeName;
+    }
+
+    /// <summary>
+    /// Gets the name of the remote computer. The server name must be ".", the local computer, when creating a server.
+    /// </summary>
+    public string ServerName { get; }
+
+    /// <summary>
+    /// Gets the name of the pipe.
+    /// </summary>
+    public string PipeName { get; }
+
+    /// <summary>
+    /// Gets the pipe name represented by this <see cref="NamedPipeEndPoint"/> instance.
+    /// </summary>
+    public override string ToString()
+    {
+        // Based on format at https://learn.microsoft.com/windows/win32/ipc/pipe-names
+        return $@"\\{ServerName}\pipe\{PipeName}";
+    }
```

Comment:
It's a little bit more complicated than this. If the app is running in a Win32 App Container (think Windows Store app, NOT docker style container), then the pipe name needs to be of the format `"\\{ServerName}\pipe\Local\{PipeName}"` as apps in a Win32 app container can't create named pipes in a non-local namespace. Or do you expect users to include the `Local\` part passed in as the pipeName?

Reply:
The user would add `Local\` to the pipe name. `PipeName` is passed directly to the pipe ctor, and it expects local as part of the name, e.g. `new NamedPipeServerStream(@"Local\MyPipeName")`